src/components/PrimaryPanes/SourcesTreeItem.js

```diff
@@ -134,17 +134,18 @@ class SourceTreeItem extends Component<Props, State> {
           click: () => copyToTheClipboard(contents.url)
         };
 
-        const blackBoxMenuItem = {
-          id: "node-menu-blackbox",
-          label: source.isBlackBoxed
-            ? L10N.getStr("sourceFooter.unblackbox")
-            : L10N.getStr("sourceFooter.blackbox"),
-          accesskey: L10N.getStr("sourceFooter.blackbox.accesskey"),
-          disabled: !shouldBlackbox(source),
-          click: () => this.props.toggleBlackBox(source)
-        };
-
-        menuOptions.push(copySourceUri2, blackBoxMenuItem);
+        if (source) {
```

Comment:
maybe we should replace the checks above `if (!Array.isArray(contents)) {` which seems like it is doing the same thing